tool/gravity/cli/top.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/gravity/lib/constants"
+	"github.com/gravitational/gravity/lib/defaults"
+	"github.com/gravitational/gravity/lib/localenv"
+	"github.com/gravitational/gravity/lib/ops"
+	"github.com/gravitational/gravity/lib/ops/monitoring"
+	"github.com/gravitational/gravity/lib/ops/opsservice"
+	"github.com/gravitational/gravity/lib/utils"
+
+	"github.com/buger/goterm"
+	"github.com/dustin/go-humanize"
+	"github.com/gizak/termui"
+	"github.com/gravitational/trace"
+)
+
+func top(env *localenv.LocalEnvironment, interval, step time.Duration) error {
+	prometheusAddr, err := utils.ResolveAddr(env.DNS.Addr(), fmt.Sprintf("%v:%v",
+		defaults.PrometheusServiceAddr, defaults.PrometheusServicePort))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	prometheusClient, err := monitoring.NewPrometheus(prometheusAddr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = termui.Init()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer termui.Close()
+
+	go render(env, context.TODO(), prometheusClient, interval, step)
+
+	termui.Handle("/sys/kbd/q", func(termui.Event) {
+		termui.StopLoop()
+	})
+	termui.Loop()
+
+	return nil
+}
+
+// render continuously spins in a loop retrieving cluster metrics and rendering
+// terminal widgets at a certain interval.
+func render(env *localenv.LocalEnvironment, ctx context.Context, client monitoring.Metrics, interval, step time.Duration) {
```

Comment:
I wonder if all this belongs in `tool/gravity/cli` and not in a separate package?

Reply:
Well, this draws in the terminal so IMO it fits well in the CLI package for now. We can move this out elsewhere if we keep developing / adding more stuff to it.